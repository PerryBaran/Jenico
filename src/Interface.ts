interface Songs {
    name: string;
    src?: string;
    ref?: string;
}

interface SongInfo {
    title: string;
    art: string;
    songs: Songs[];
    hyperlink: string;
}


export type {Songs, SongInfo}