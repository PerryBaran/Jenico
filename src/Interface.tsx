interface Songs {
    name: string;
    src: string;
}

interface SongInfo {
    title: string;
    art: string;
    songs: Songs[];
    hyperlink: string;
}


export type {Songs, SongInfo}