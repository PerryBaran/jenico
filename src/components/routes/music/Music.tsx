import { Link } from "react-router-dom";
import { SongInfo } from "../../../Interface";
import style from "./music.module.css";
import Background from "../../background/Background";
import { background } from "../../../media/images/index";

function Music(props: { data: SongInfo[] }) {
  const { data } = props;

  return (
    <>
      <Background src={background} />
      <div className={style.container}>
        {data.map((album: SongInfo) => {
          return (
            <Link to={album.title} key={album.title} className={style.album}>
              <img src={album.art} alt={`${album.title} album cover`} />
              <p>{album.title}</p>
            </Link>
          );
        })}
      </div>
    </>
  );
}

export default Music;
