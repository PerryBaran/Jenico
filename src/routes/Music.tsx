import React from 'react';

function Music(props:any) {
    const {data} = props
    return (
        <div className="music">
            {data.map(((album: any) => {
                return (
                    <button className='cover'>
                        <div>
                            <img src={album.art} alt={`${album.title} album cover`}/>
                            <p>{album.title}</p>
                        </div>
                    </button>
                )
            }))}
        </div>
    );
}

export default Music;