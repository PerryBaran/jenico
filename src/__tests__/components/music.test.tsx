import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Music from "../../components/routes/music/Music";
import mockData from "../../media/songInfo";
import { SongInfo } from "../../Interface";
import { BrowserRouter as Router } from "react-router-dom";

const RenderWithRouter = (props: { data: SongInfo[] }) => {
  const { data } = props;

  return (
    <Router>
      <Music data={data} />
    </Router>
  );
};

describe("Music", () => {
  const validProps = {
    data: mockData,
  };

  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(<RenderWithRouter data={validProps.data} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("content", () => {
    render(<RenderWithRouter data={validProps.data} />);

    expect(screen.getByAltText(/background/)).toBeInTheDocument();

    const links = screen.getAllByRole("link");
    links.forEach((link, i) => {
      const album = validProps.data[i];

      expect(link).toHaveAttribute("href", `/${album.title}`);
      expect(screen.getByAltText(`${album.title} album cover`)).toHaveAttribute(
        "src",
        album.art
      );
      expect(screen.getByText(album.title)).toBeInTheDocument();
    });
  });
});
