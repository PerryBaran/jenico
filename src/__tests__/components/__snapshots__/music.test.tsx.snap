// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="background"
      src="background.jpg"
    />
  </div>
  <div
    class="container"
  >
    <a
      class="album"
      href="/Where We End"
    >
      <img
        alt="Where We End album cover"
        src="WhereWeEnd.png"
      />
      <p>
        Where We End
      </p>
    </a>
    <a
      class="album"
      href="/Dreaming of Detuned Love"
    >
      <img
        alt="Dreaming of Detuned Love album cover"
        src="DreamingOfDetunedLove.png"
      />
      <p>
        Dreaming of Detuned Love
      </p>
    </a>
    <a
      class="album"
      href="/Lysergic"
    >
      <img
        alt="Lysergic album cover"
        src="Lysergic.png"
      />
      <p>
        Lysergic
      </p>
    </a>
    <a
      class="album"
      href="/Ethereal"
    >
      <img
        alt="Ethereal album cover"
        src="Ethereal.png"
      />
      <p>
        Ethereal
      </p>
    </a>
    <a
      class="album"
      href="/Storm EP"
    >
      <img
        alt="Storm EP album cover"
        src="StormEP.png"
      />
      <p>
        Storm EP
      </p>
    </a>
  </div>
</DocumentFragment>
`;
